src/experimental-bots/spotFiller/spotFillerMultithreaded.ts

```diff
@@ -1767,11 +1776,18 @@ export class SpotFillerMultithreaded {
 								txAge / 1000
 							} s`
 						);
+						logger.debug(
+							`Removing confirmed txSig ${txSig} from pendingTxSigsToconfirm, new size: ${this.pendingTxSigsToconfirm.size}`
+						);
 						this.pendingTxSigsToconfirm.delete(txSig);
 						if (txType === 'fill') {
+							let node: any = nodeFilled;
+							if (Array.isArray(nodeFilled)) {
+								node = nodeFilled[0];
+							}
```

Comment:
we should remove this, given that it should never be an Array. we should just error if it's not the type that we expect